File: voice-note-server/src/main/kotlin/com/suikouji/server/llm/dto/TransactionParseResponse.kt
package com.suikouji.server.llm.dto

import com.fasterxml.jackson.annotation.JsonProperty

/** Structured transaction data extracted by LLM from natural language. */
data class TransactionParseResponse(
    val amount: Double?,
    val currency: String = "CNY",
    val date: String?,
    val category: String?,
    val description: String?,
    val type: TransactionType?,
    val account: String?,
    @JsonProperty("transfer_direction") val transferDirection: String? = null,
    val counterparty: String? = null,
    val confidence: Double = 0.0,
    val model: String = ""
)

enum class TransactionType {
    EXPENSE, INCOME, TRANSFER
}

File: voice-note-client/lib/core/network/dto/transaction_parse_response.dart
/// Response from POST /api/v1/llm/parse-transaction.
class TransactionParseResponse {
  final double? amount;
  final String currency;
  final String? date;
  final String? category;
  final String? description;
  final String? type;
  final String? account;
  final String? transferDirection;
  final String? counterparty;
  final double confidence;
  final String model;

  const TransactionParseResponse({
    this.amount,
    this.currency = 'CNY',
    this.date,
    this.category,
    this.description,
    this.type,
    this.account,
    this.transferDirection,
    this.counterparty,
    required this.confidence,
    required this.model,
  });

  factory TransactionParseResponse.fromJson(Map<String, dynamic> json) {
    return TransactionParseResponse(
      amount: (json['amount'] as num?)?.toDouble(),
      currency: json['currency'] as String? ?? 'CNY',
      date: json['date'] as String?,
      category: json['category'] as String?,
      description: json['description'] as String?,
      type: json['type'] as String?,
      account: json['account'] as String?,
      transferDirection: json['transfer_direction'] as String?,
      counterparty: json['counterparty'] as String?,
      confidence: (json['confidence'] as num?)?.toDouble() ?? 0.0,
      model: json['model']?.toString() ?? '',
    );
  }

  /// Whether all required fields for a transaction are present.
  bool get isComplete => amount != null && category != null;
}
